src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -1,32 +1,85 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
 
+import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.ApplicationListener
 import org.springframework.stereotype.Service
 import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.EmailSender
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ActionPlanEvent
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ActionPlanEventType
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
-import uk.gov.service.notify.NotificationClient
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
 import java.net.URI
 import java.util.UUID
 
+interface NotifyService {
+  fun generateResourceUrl(baseURL: String, path: String, id: UUID): URI {
+    return UriComponentsBuilder.fromHttpUrl(baseURL).path(path).buildAndExpand(id).toUri()
+  }
+
+  fun getUserDetails(user: AuthUser, hmppsAuthService: HMPPSAuthService, log: Logger) =
+    try {
+      hmppsAuthService.getUserDetail(user)
+    } catch (e: Exception) {
+      log.error("could not get account details for assigned service provider user", e)
+      null
+    }
+}
+
+@Service
+class NotifyActionPlanService(
+  @Value("\${notify.templates.action-plan-submitted}") private val actionPlanSubmittedTemplateID: String,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.submit-action-plan}") private val interventionsUISubmitActionPlanLocation: String,
+  private val emailSender: EmailSender,
+  private val hmppsAuthService: HMPPSAuthService,
+) : ApplicationListener<ActionPlanEvent>, NotifyService {
+
+  override fun onApplicationEvent(event: ActionPlanEvent) {
+    when (event.type) {
+
+      ActionPlanEventType.SUBMITTED -> {
+        getUserDetails(event.actionPlan.submittedBy!!, hmppsAuthService, log)?.let {
+          val location = generateResourceUrl(interventionsUIBaseURL, interventionsUISubmitActionPlanLocation, event.actionPlan.referral.id)
+          emailSender.sendEmail(
+            actionPlanSubmittedTemplateID,
+            "rajiv.mahal@digital.justice.gov.uk",
```

Comment:
I'm assuming this is just temporary, but a note to remember to update this :)

Reply:
Good spot. Corrected